SecretSanta/src/test/SecretSanta.Api.Tests/Controllers/UsersControllerTests.cs

```diff
@@ -0,0 +1,218 @@
+        public void Constructor_WithNullUserManager_ThrowsAppropriateException()
+        {
+            ArgumentNullException ex = Assert.ThrowsException<ArgumentNullException>(
+                () => new UsersController(null!));
+
+            try
+            {
+                new UsersController(null!);
+            }
+            catch (ArgumentNullException e)
+            {
+                Assert.AreEqual("userManager", e.ParamName);
+                return;
+            }
+            Assert.Fail("No exception thrown");
+        }
+
+        // GET
+
+        [TestMethod]
+        public void Get_WithData_ReturnsUsers()
+        {
+            //Arrange
+            UsersController controller = new(new UserManager());
+
+            //Act
+            IEnumerable<User> users = controller.Get();
+
+            //Assert
+            Assert.IsTrue(users.Any());
+        }
+
+        [TestMethod]
+        [DataRow(42)]
+        [DataRow(98)]
+        public void Get_WithId_ReturnsUserManagerUser(int id)
+        {
+            //Arrange
+            TestableUserManager manager = new();
+            UsersController controller = new(manager);
+            User expectedUser = new();
+            manager.GetItemUser = expectedUser;
+
+            //Act
+            ActionResult<User?> result = controller.Get(id);
+
+            //Assert
+            Assert.AreEqual(id, manager.GetItemId);
+            Assert.AreEqual(expectedUser, result.Value);
+        }
+
+        [TestMethod]
+        public void Get_WithNegativeId_ReturnsNotFound()
+        {
+            //Arrange
+            TestableUserManager manager = new();
+            UsersController controller = new(manager);
+            User expectedUser = new();
+            manager.GetItemUser = expectedUser;
+
+            //Act
+            ActionResult<User?> result = controller.Get(-1);
+
+            //Assert
+            Assert.IsTrue(result.Result is NotFoundResult);
+        }
+
+
+        // DELETE 
+        [TestMethod]
+        [DataRow(1)]
+        public void Delete_WithId_OkReturned(int id)
+        {
+            //Arrange
+            UserManager manager = new();
+            UsersController controller = new(manager);
+
+            //Act
+            ActionResult result = controller.Delete(id);
+
+            //Assert
+            Assert.IsTrue(result is OkResult);
+        }
+
+        [TestMethod]
+        [DataRow(-4000)]
+        public void Delete_WithId_NotFoundReturned(int id)
+        {
+            //Arrange
+            UserManager manager = new();
+            UsersController controller = new(manager);
+
+            //Act
+            ActionResult result = controller.Delete(id);
+
+            //Assert
+            Assert.IsTrue(result is NotFoundResult);
+        }
+
+        // POST
+        [TestMethod]
+        [DataRow(200, "PostTestFirst", "PostTestLast")]
+        public void Post_WithUserData_ConfrimUpdatedWithGet(int id, String first, String last)
```

Comment:
Nit: When available favor using C# keywords rather than Type names when using primitives (int, double, string, bool, etc). 
```suggestion
        public void Post_WithUserData_ConfrimUpdatedWithGet(int id, string first, string last)
```